src/components/FilterBar.jsx

```diff
@@ -40,13 +40,27 @@ export default function FilterBar(props) {
   const {
     changeSortBy,
     changeSortFormat,
-    onAmountFilterChange,
+    setQueryParam,
+    clearQueryParam,
     queryParams,
   } = props;
   const classes = useStyles();
 
-  const minAmount = queryParams.minAmount ?? 0;
-  const maxAmount = queryParams.maxAmount ?? 0;
+  const { minAmount, maxAmount } = queryParams;
+
+  if (
+    minAmount !== undefined &&
+    !(Number.isInteger(minAmount) && minAmount > 0)
+  ) {
+    clearQueryParam('minAmount');
```

Comment:
We should probably refactor these query param magic strings somewhere. Maybe a constant called `QUERY_PARAM_MIN_AMOUNT` or `QueryParams.MIN_AMOUNT` where there's some `src/lib/QueryParams.js` with a `MIN_AMOUNT` constant?